### PR TITLE
issues:bug fixes

### LINKED ISSUE #56: Get issue returns entire document
In the current implementation, when a request for issue(s) is sent to the API, the API returns entire mongodb document, which is not desirable. Only the required must be sent.

### PR DESCRIPTION
**Pull request by FaiyazMujawar**
Fixes #56 

In this PR,
- The get requests for the issues now return only the required properties as per the discussion.
- A small bug in user roles was fixed. The authority didnt had access to read any issue.

### CODE REVIEW

File: controllers/IssueController.js [hunk 1]
```diff
@@ -459,6 +463,28 @@ const deleteIssue = async (req, res) => {
...
+
+  filteredIssue["isAuthor"] = filteredIssue.createdBy.toString() === userId;
+  filteredIssue["hasUpvoted"] = !!filteredIssue.upvoters.find(
+    (id) => id.toString() === userId
+  );
+
+  delete filteredIssue.__v;
+  delete filteredIssue.isDeleted;
+  delete filteredIssue.createdBy;
+  delete filteredIssue.upvoters;
```
Comment (drigio) [inline controllers/IssueController.js:483]: Why are we deleting count of upvoters to be returned back?

Comment (drigio) [reply controllers/IssueController.js:483]: My bad I didn't see that we were sending `upvotes` back in the response. I'll mark it as resolved.

**Comment by drigio** [comment]
LGTM! I can merge the changes, if you don't have any further changes.

**Comment by FaiyazMujawar** [comment]
Go ahead if you think it doesn't need any changes.

**Pull request merged**